rust/extractor/src/translate/base.rs

```diff
@@ -26,12 +26,33 @@ use ra_ap_syntax::{
 macro_rules! pre_emit {
     (Item, $self:ident, $node:ident) => {
         if let Some(label) = $self.prepare_item_expansion($node) {
-            return Some(label);
+            return Some(label.into());
+        }
+    };
+    (AssocItem, $self:ident, $node:ident) => {
+        if let Some(label) = $self.prepare_item_expansion(&$node.clone().into()) {
+            return Some(label.into());
+        }
+    };
+    (ExternItem, $self:ident, $node:ident) => {
+        if let Some(label) = $self.prepare_item_expansion(&$node.clone().into()) {
+            return Some(label.into());
         }
     };
     ($($_:tt)*) => {};
 }
-
+impl From<crate::trap::Label<generated::AssocItem>> for crate::trap::Label<generated::Item> {
```

Comment:
please let's put in a TODO that this is temporary while we look for a solution going through the generated schema